packages/container/lib/container.js

```diff
@@ -299,18 +321,40 @@ function instantiate(container, fullName) {
 
     validationCache[fullName] = true;
 
+    let obj;
+
     if (typeof factory.extend === 'function') {
       // assume the factory was extendable and is already injected
-      return factory.create();
+      obj = factory.create();
     } else {
       // assume the factory was extendable
       // to create time injections
       // TODO: support new'ing for instantiation and merge injections for pure JS Functions
-      return factory.create(injectionsFor(container, fullName));
+      obj = factory.create(injectionsFor(container, fullName));
+
+      if (isEnabled('ember-container-inject-owner')) {
+        injectDeprecatedContainer(obj, container);
```

Comment:
No, we provide the actual (non deprecated) container in `injectionsFor`, then change it to the deprecated one. This unfortunately means that when the feature is enabled any non Ember Objects instantiated by the container (since this is in the `else` clause here) will receive the non-deprecated `container` during their `create` call, then get `.container` swapped out to the deprecated one after instantiation.

The solution to this is to create a complete `FakeContainer` (as mentioned in a comment above and in some of the PR comments).  That change is definitely a blocker to this feature being enabled, but I would like to do it as a separate (and smaller PR).